Source/Core/VideoBackends/OGL/VertexManager.cpp

```diff
@@ -140,6 +140,15 @@ void VertexManager::vFlush(bool useDstAlpha)
 
 	PrepareDrawBuffers(stride);
 
+	if (!bpmem.genMode.zfreeze && IndexGenerator::GetIndexLen() >= 3)
+	{
+		CalculateZSlope(stride);
```

Comment:
@Tinob that's not accurate to hardware though - the point is exactly that the zslope of the last triangle is used before zfreeze was enabled.

Assuming that games render a reference plane with zfreeze already enabled is just asking for trouble.